rfd/xxxx-automatic-azure-server-discovery.md

```diff
@@ -0,0 +1,146 @@
+later.
+
+## Why
+
+RFD 57 replaced the process of manually installing Teleport on EC2 servers, which
+could be slow for large numbers of instances. This RFD adds discovery on Azure
+instances. The installation of Teleport on these servers is left to a future RFD.
+
+## Details
+
+### Discovery
+
+Azure discovery will be handled by the new [discovery service](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md#discovery) described in RFD 57.
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  azure:
+    - types: ["vm"]
+      regions: ["westcentralus"]
+      tags:
+        "teleport": "yes"
+```
+
+The Teleport discovery service will need a [service principal](https://learn.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest) with a role that includes the `Microsoft.Compute/virtualMachines/read`
+permission to list virtual machines via the Go Azure SDK.
+
+As with AWS database discover and EC2 discover, new Azure nodes will be discovered
+periodically on a 60 second timer, as new nodes are found they will be added to the
+teleport cluster.
+
+In order to avoid attempting to reinstall Teleport on top of an instance where it is
+already present, the generated Teleport config will match against the node name using
+the Azure subscription ID and the virtual machine ID.
+
+```json
+{
+  "kind": "node",
+  "version": "v2",
+  "metadata": {
+    "name": "${AZURE_SUBSCRIPTION_ID}-${VM_ID}",
+    "labels": {
+      "env": "example",
+      "teleport.dev/discovered-node": "yes",
+      "teleport.dev/discovered-by": "${DISCOVER_NODE_UUID}",
+      "teleport.dev/origin": "cloud",
+      "teleport.dev/region": "westcentralus",
+      "teleport.dev/subscriptionId": "88888888-8888-8888-8888-888888888888"
+    }
+  },
+  "spec": {
+    "public_addr": "...",
+    "hostname": "azurexyz"
+  }
+}
+```
+
+Alternatively, if the instance has the tag `teleport.dev/instance_name` present, the tag
+value will override the node name.
+
+```json
+{
+  "kind": "node",
+  "version": "v2",
+  "metadata": {
+    "name": "custom_node_name",
+    "labels": {
+      "teleport.dev/discovered-node": "yes",
+      "teleport.dev/discovered-by": "${DISCOVER_NODE_UUID}",
+      "teleport.dev/origin": "cloud",
+      "teleport.dev/instance_name": "custom_node_name",
+      "teleport.dev/region": "westcentralus",
+      "teleport.dev/subscriptionId": "88888888-8888-8888-8888-888888888888"
+    }
+  },
+  "spec": {
+    "public_addr": "...",
+    "hostname": "azurexyz"
+  }
+}
+```
+
+### Joining
+
+Until joining is implemented, the discovery service will simply log the presence of
+discovered servers.
+
+## UX
+
+### User has 1 account to discover servers on
+
+#### Teleport config
+
+Discovery server:
+```yaml
+teleport:
+  ...
+auth_service:
+  enabled: "yes"
+discovery_service:
+  enabled: "yes"
+  azure:
+    - types: ["vm"]
```

Comment:
Include all fields defined in [AzureMatcher ](https://github.com/gravitational/teleport/blob/09773205f7fd96541e5d98ef0d2a790a8317a84e/lib/config/fileconf.go#L1357)
This is just to avoid confusion if anyone reads the RFD and doesn't see all the fields supported in the example.